server.js

```diff
@@ -1,23 +1,104 @@
 const express = require("express");
-const cors = require("cors");
-
 const app = express();
+app.use(express.json());
 
+const cors = require("cors");
 app.use(cors());
 
+app.use(express.urlencoded({ extended: false }));
+
+app.get("/", function (request, response) {
+  response.sendFile(__dirname + "/index.html");
+});
+
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const myMessage = {
+  id: 1,
+  from: "Jas",
+  text: "Hello, how are you?",
+};
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+const joannaMessage = {
+  id: 2,
+  from: "Joanna",
+  text: "Have a lovely day",
+};
+
+const messages = [welcomeMessage, myMessage, joannaMessage];
+
+//create message
+
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+
+  const newMessage = {
+    id: messages.length,
+    from,
+    text,
+    timeStamp: (TimeDate = new Date()),
+  };
+
+  if (!newMessage.from || !newMessage.text) {
+    return res.status(400).json("Please include a name and message");
+  }
+  messages.push(newMessage);
+  res.send(messages);
+});
+//read all messages
+app.get("/messages", (req, res) => {
+  res.send(messages);
 });
 
-app.listen(process.env.PORT);
+//Read one message specified by an ID
+app.get("/messages/:ID", (req, res) => {
+  const { ID } = req.params;
+  const messageWithId = messages.find((elem) => elem.id == ID);
+  res.send(messageWithId);
+});
+
+//Delete a message, by ID
+app.delete("/messages/:ID", (req, res) => {
+  const ID = req.params.ID;
+  const notDeletedMessage = messages.filter((elem) => elem.id != ID);
+  res.send(notDeletedMessage);
+});
+
+//read only messages with given substring
+app.get("/messages/search", (req, res) => {
+  let messageRead = messages.filter((message) =>
+    message.text.includes(req.query.text)
+  );
+  res.send(messageRead);
+});
+
+//read the latest messages
+app.get("/messages/latest", (req, res) => {
+  if(messages.length >= 2){
+    return res.send(messages.slice(messages.length - 2));
+  } else {
```

Comment:
You don't need the else here, as the upper conditional block ends in a return.

```
if (x >1) {
  return something
}
return somethingElse
```
looks cleaner.